extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesAndWorkerTaskRunner.java

```diff
@@ -0,0 +1,266 @@
+    taskSlotCounts.putAll(workerTaskRunner.getIdleTaskSlotCount());
+    return taskSlotCounts;
+  }
+
+  @Override
+  public Map<String, Long> getUsedTaskSlotCount()
+  {
+    Map<String, Long> taskSlotCounts = new HashMap<>();
+    taskSlotCounts.putAll(kubernetesTaskRunner.getUsedTaskSlotCount());
+    taskSlotCounts.putAll(workerTaskRunner.getUsedTaskSlotCount());
+    return taskSlotCounts;
+  }
+
+  @Override
+  public Map<String, Long> getLazyTaskSlotCount()
+  {
+    Map<String, Long> taskSlotCounts = new HashMap<>();
+    taskSlotCounts.putAll(kubernetesTaskRunner.getLazyTaskSlotCount());
+    taskSlotCounts.putAll(workerTaskRunner.getLazyTaskSlotCount());
+    return taskSlotCounts;
+  }
+
+  @Override
+  public Map<String, Long> getBlacklistedTaskSlotCount()
+  {
+    Map<String, Long> taskSlotCounts = new HashMap<>();
+    taskSlotCounts.putAll(kubernetesTaskRunner.getBlacklistedTaskSlotCount());
+    taskSlotCounts.putAll(workerTaskRunner.getBlacklistedTaskSlotCount());
+    return taskSlotCounts;
+  }
+
+  @Override
+  public Collection<ImmutableWorkerInfo> getWorkers()
+  {
+    return workerTaskRunner.getWorkers();
+  }
+
+  @Override
+  public Collection<Worker> getLazyWorkers()
+  {
+    return workerTaskRunner.getLazyWorkers();
+  }
+
+  @Override
+  public Collection<Worker> markWorkersLazy(Predicate<ImmutableWorkerInfo> isLazyWorker, int maxWorkers)
+  {
+    return workerTaskRunner.markWorkersLazy(isLazyWorker, maxWorkers);
+  }
+
+  @Override
+  public WorkerTaskRunnerConfig getConfig()
+  {
+    return workerTaskRunner.getConfig();
+  }
+
+  @Override
+  public Collection<Task> getPendingTaskPayloads()
+  {
+    return workerTaskRunner.getPendingTaskPayloads();
+  }
+
+  @Override
+  public Optional<InputStream> streamTaskLog(String taskid, long offset) throws IOException
+  {
+    Optional<InputStream> kubernetesTaskLog = kubernetesTaskRunner.streamTaskLog(taskid, offset);
+    if (kubernetesTaskLog.isPresent()) {
+      return kubernetesTaskLog;
+    } else if (workerTaskRunner instanceof TaskLogStreamer) {
+      return ((TaskLogStreamer) workerTaskRunner).streamTaskLog(taskid, offset);
+    }
+    return Optional.absent();
+  }
+
+  @Nullable
+  @Override
+  public RunnerTaskState getRunnerTaskState(String taskId)
+  {
+    RunnerTaskState runnerTaskState = kubernetesTaskRunner.getRunnerTaskState(taskId);
+    if (runnerTaskState == null) {
+      return workerTaskRunner.getRunnerTaskState(taskId);
+    }
+
+    return runnerTaskState;
+  }
+
+  @Override
+  public List<TaskRunner> getSubTaskRunners()
+  {
+    return ImmutableList.of(kubernetesTaskRunner, workerTaskRunner);
+  }
+
+  @Override
+  public int getTotalCapacity()
+  {
+    return kubernetesTaskRunner.getTotalCapacity() + workerTaskRunner.getTotalCapacity();
+  }
+
+  @Override
+  public int getUsedCapacity()
+  {
+    return kubernetesTaskRunner.getUsedCapacity() + workerTaskRunner.getUsedCapacity();
+  }
```

Comment:
The interface definitions say used capacity and total capacity can return -1 to indicate it is not implemented / tasks can not be found. It seems like there should be some special handling here if either task runner returns -1.

However, looking at the implementations of the 3 task runners involved, it doesn't look like any of them return -1 ever.